engine/src/engine.rs

```diff
@@ -1243,6 +1296,14 @@ pub fn get_balance<I: IO>(io: &I, address: &Address) -> Wei {
     Wei::new(raw)
 }
 
+/// Checks whether there are enough funds for the given address and amount.
+///
+/// This does not take gas into account.
+pub fn can_transfer<I: IO>(io: &I, address: &Address, amount: Wei) -> bool {
+    let current_balance = get_balance(io, address);
+    amount >= current_balance
```

Comment:
Probably should be less or equal?
 
```suggestion
    amount <= current_balance
```